### ISSUE TITLE
The component Styled(IframeResizer) with the id of "sc-bdnylx" has been created dynamically. You may see this warning because you've called styled inside another component. To resolve this only create new StyledComponents outside of any render method and function component.

### ISSUE THREAD

**Issue by RickGove** [styled-components/styled-components#3290]
### I've searched for the answer already, and I apologize if this is formatted incorrectly...

I've looked at a similar issue on GitHub to this one. However, I'm not creating my `Styled-Component` inside of a function component, so I have no idea what is going wrong. This is also not my code, I am just trying to get something working...

### Problems with loading style theme from context

The issue I'm trying to fix is that some components are updating their style based on the theme in `React.context`. I believe that this could be the issue. I have ten of these warnings on the actual code...

### Here is my warning:

```error
The component Styled(IframeResizer) with the id of "sc-bdnylx" has been created dynamically.
You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.
```

### The code:

```jsx
import React, { useRef, useEffect, useState } from 'react'
import IframeResizer from 'iframe-resizer-react'
import styled from 'styled-components'
import useMessage from '../use-message'

const StyledIFrame = styled(IframeResizer)`
  border: none;
  width: 100%;
`

export const IFrameHost = (props) => {
  const { notebook, onChange, documentId } = props
  const iframeRef = useRef()
  const srcRef = useRef()

  const [isSourceReady, setSourceReady] = useState()
  const [isBootstrapComplete, setBootstrapComplete] = useState()
  const [isNotebookReady, setNotebookReady] = useState()

  const onMessage = (event, message) => {
    if (message?.type === 'document-updated') {
      onChange?.(message.payload)
    }
    if (message?.type === 'bootstrap-complete') {
      setBootstrapComplete(true)
      // console.log('🥳🥳 BOOTSTRAP COMPLETE! 🥳🥳')
    }
  }

  const messaging = useMessage({
    isHost: true,
    srcRef: srcRef,
    onMessage
  })

  const onResized = () => ({})

  useEffect(() => {
    if (
      iframeRef.current?.iframeRef?.current &&
      (!srcRef.current ||
        srcRef.current !== iframeRef.current?.iframeRef?.current)
    ) {
      srcRef.current = iframeRef.current?.iframeRef?.current
      srcRef.current.onload = (k) => {
        srcRef.current = k.target
        setSourceReady((v) => {
          if (v) {
            setTimeout(() => setSourceReady(true), 2)
            return false
          }
          return true
        })
      }
    }
  }, [iframeRef, isSourceReady, srcRef, iframeRef.current])

  useEffect(() => {
    srcRef.current = null
  }, [documentId])

  const setFailedToBootstrap = () => {
    if (!isBootstrapComplete) {
      console.log('±±± BOOTSTRAP FAILED ±±±')
    } else {
      console.log('±±± BOOTSTRAP SUCCEEDED ±±±')
    }
  }

  useEffect(() => {
    if (isSourceReady && !isBootstrapComplete) {
      // console.log('±±± BOOTSTRAP START ±±±')
      messaging.sendMessage({ type: 'bootstrap', testing: 'xup' })
      const tid = setTimeout(() => {
        // console.log('±±± BOOTSTRAP TIMEOUT CHECK ±±±')
        setFailedToBootstrap(true)
      }, 1000)
      return () => {
        // console.log('±±± BOOTSTRAP TIMEOUT CHECK CANCELED ±±±')
        clearTimeout(tid)
      }
    }
  }, [isSourceReady, isBootstrapComplete])

  useEffect(() => {
    if (notebook) {
      setNotebookReady(true)
    }
  }, [notebook])

  useEffect(() => {
    if (isNotebookReady && isSourceReady)
      messaging.sendMessage({ type: 'notebook-updated', notebook })
  }, [isNotebookReady, isSourceReady])

  return (
    <StyledIFrame
      heightCalculationMethod='taggedElement'
      forwardRef={iframeRef}
      src={props.src}
      onMessage={onMessage}
      onResized={onResized}
    />
  )
}

export default IFrameHost

```

**Comment by kitten** [comment]
Have you checked whether you duplicate React packages in the environment where you're testing this in?

Also quick note that you search for an answer you can often trim down the message and search our closed issues. I think there were prior issues with the same question 🤔

**Comment by RogueXiaoxi** [comment]
这是ul框架么？

**Comment by capndave** [comment]
@RickGove how was this resolved? I have the same issue

**Comment by kitten** [comment]
Closing due to inactivity.

Please don't forget to be a good netizen on GitHub and revisit issues, especially ones that have been resolved and need to be closed, or ones that have received replies ❤️
